go-peer/main.go

```diff
@@ -144,13 +156,45 @@ func main() {
 		panic(err)
 	}
 
-	// create a new libp2p Host that listens on a random TCP port
-	h, err := libp2p.New(
+	// TLS stuff
+	var opts []libp2p.Option
+
+	if *certPath != "" && *keyPath != "" {
+		certs := make([]tls.Certificate, 1)
+		certs[0], err = tls.LoadX509KeyPair(*certPath, *keyPath)
+		if err != nil {
+			panic(err)
+		}
+
+		opts = append(opts,
+			libp2p.Transport(ws.New, ws.WithTLSConfig(&tls.Config{Certificates: certs})),
+			libp2p.ListenAddrStrings("/ip4/0.0.0.0/tcp/0/ws", "/ip6/::/tcp/0/ws"),
+		)
+	}
+
+	if len(announceAddrs) > 0 {
+		var announce []multiaddr.Multiaddr
+		for _, addr := range announceAddrs {
+			announce = append(announce, multiaddr.StringCast(addr))
+		}
+		opts = append(opts, libp2p.AddrsFactory(func([]multiaddr.Multiaddr) []multiaddr.Multiaddr {
+			return announce
+		}))
+	}
+
+	opts = append(opts,
 		libp2p.Identity(privk),
 		libp2p.Transport(quicTransport.NewTransport),
+		libp2p.Transport(tcpTransport.NewTCPTransport),
 		libp2p.Transport(webtransport.New),
-		libp2p.ListenAddrStrings("/ip4/0.0.0.0/udp/0/quic-v1", "/ip4/0.0.0.0/udp/0/quic-v1/webtransport", "/ip6/::/udp/0/quic-v1", "/ip6/::/udp/0/quic-v1/webtransport"),
+		libp2p.ListenAddrStrings("/ip4/0.0.0.0/udp/0/quic-v1", "/ip4/0.0.0.0/udp/0/quic-v1/webtransport", "/ip6/::/udp/0/quic-v1", "/ip6/::/udp/0/quic-v1/webtransport",
+			"/ip4/0.0.0.0/udp/0/quic", "/ip6/::/udp/0/quic",
+			"/ip4/0.0.0.0/tcp/0", "/ip6/::/tcp/0",
```

Comment:
```suggestion
		libp2p.ListenAddrStrings("/ip4/0.0.0.0/udp/9091/quic-v1", "/ip4/0.0.0.0/udp/9092/quic-v1/webtransport", "/ip4/0.0.0.0/tcp/0"),
```
we don't need to support IPv6 or QUIC v0 I think
Also changed made it run on port 90

Reply:
Agreed regarding QUIC draft versions (it's not called v0). Why not IPv6 though?

Reply:
> Why not IPv6 though?

The other peers aren't listening on v6 so might as well disable it here

Reply:
FWIW ipv6 is usually easier to peer with (no NAT), but I'm neutral otherwise if the other nodes aren't going to support it

Reply:
Not that it matters much here, but I'd be a bit sad building something in 2023 that doesn't do IPv6.